package proxy.msg;

enum BIG_CMD {
    ServerCmd = 0x500;
}

enum SUB_CMD {
    SubCmd0x1 = 0x1; //connect msg
    SubCmd0x2 = 0x2; //data msg
}

message ProxyMsg {
    optional bytes place_holder = 1;    //占位符, 任意长度
    optional uint32 check_sum = 2;      //校验值, body的校验
    optional uint32 cmd = 3;            //大命令字
    optional uint32 subcmd = 4;         //小命令字
    optional uint32 seq = 5;            //序号
    required uint32 err_code = 6;       //errcode
    optional bytes body = 10;            //body
}

message ProxyConnectReq {
    optional uint32 type = 1; //1-ipv4, 3-host, 4-ipv6
    optional string address = 2; //由type指定
    optional uint32 port = 3;     //port
    optional string username = 4; //用户名
    optional string password = 5; //密码
}

message ProxyConnectRsp {
    required uint32 result = 1; //0-ok, other-err
}


